SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityJobPoller.java

```diff
@@ -69,6 +72,16 @@ public void runActionOnPoll() {
         continue;
       }
 
+      if (start - taskId.getStartedAt() >= configuration.getTaskExecutionTimeout()) {
```

Comment:
So the `SCHEDULED` tasks should be shut down as well. Right now, the method just warns that it's overdue. Should that part be removed since it's redundant or should `SCHEDULED` request types just continue to have overdue alerts while `ON_DEMAND` and `RUN_ONCE` tasks are cleaned up?